connection.go

```diff
@@ -103,11 +107,21 @@ Loop:
 				}
 			}()
 
+			go func() {
+				for _ = range setTimeoutChan {
+				}
+			}()
+
 			i.writeLock.Lock()
```

Comment:
writeLock is only used here and in WriteFrame. setTimeoutLock is only used here and in SetIdleTimeout. Each lock separately guards against concurrent access that would lead to an NPE for the item in question (i.resetChan for writeLock and i.setTimeoutChan for setTimeoutLock). I had considered trying to use one lock for both of them, but that seemed questionable. I don't believe there is any issue doing it this way, but maybe we should think through some scenarios.